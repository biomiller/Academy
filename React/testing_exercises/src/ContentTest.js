import React, { Component } from 'react';

export class ContentTest extends Component {

    constructor() {
        super();
        this.state = {
            count: 0
        }
    }



    tickUp = () => {
        this.setState({
            count: this.state.count + 1
        })
    }
    render() {
        return (
            <div>
                <h2>Look at me i'm content!</h2>
                <button name="addOne" onClick={this.tickUp}>{this.state.count}</button>
            </div>
        );
    }

}
