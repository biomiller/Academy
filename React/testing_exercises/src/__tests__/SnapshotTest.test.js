import React from 'react';
import { shallow } from 'enzyme';
import { ContentTest } from '../ContentTest';

it('snapshot test', () => {
  const wrapper = shallow(<ContentTest />);
  
  expect(wrapper).toMatchSnapshot()
  
  wrapper.find('[name="addOne"]').simulate("click");

  expect(wrapper).toMatchSnapshot();

});
