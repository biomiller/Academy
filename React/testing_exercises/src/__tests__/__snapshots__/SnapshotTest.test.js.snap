// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `ShallowWrapper {}`;

exports[`snapshot test 2`] = `ShallowWrapper {}`;
